package.esm.json

```diff
@@ -0,0 +1,3 @@
+{
+  "type": "module"
```

Comment:
Minor: I'd rather see this file checked in under `dist/esm` than placed here and copied at build time. That matches what's happening in the 3.0.0 branch.

Reply:
I assumed it was possible for dist to have `rm -rf` ran against it before the build, so that's why I opted for copying. Easy to change.

Reply:
@jasonpaulos Updated the PR to incorporate this feedback.